forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -1888,6 +1889,18 @@ public final void setChosenPlayer(final Player p) {
         view.updateChosenPlayer(this);
     }
 
+    public final boolean hasPromisedGift() {
+        return promisedGift != null;
+    }
+    public final Player getPromisedGift() {
+        return promisedGift;
+    }
+    public final void setPromisedGift(final Player p) {
```

Comment:
this needs to be called in `copySpellHost`

Reply:
Lemme know if thats what you meant